src/main.go

```diff
@@ -284,9 +334,114 @@ func garbageClean(directory, cname, current_version string, size_wanted int64) e
 	return nil
 }
 
-const ERR_INVALID_ARGUMENTS = 1
-const ERR_SYSTEM_FAILURE = 2
-const ERR_NETWORK_PROBLEMS = 3
+func verifyManifest(repo *remote.Repository, ctx context.Context, digest, verificationKeyFile string) {
+	signatureTag := strings.Replace(digest, "sha256:", "sha256-", 1) + ".sig"
```

Comment:
shouldn't we ensure to only replace at start of string? or even simpler, if we assume the string always starts with "sha256", then just do `signatureTag := digest` `signatureTag[6] = "-"`

Reply:
I think this is a pretty safe way to convert the hash-string to the tag. You are right, "sha256:" is not the shortest match-pattern that one can choose nor is it the quickest way to transform the string, but I think this has a good readability. You have to dig a little bit into the cosign docs to find this conversion mechanism and if we use that code here it is a little bit more self-explanatory from my point of view.